model-optimizer/unit_tests/extensions/back/ShapeOfConstFolding_test.py

```diff
@@ -0,0 +1,86 @@
+                    'new_const_shapeof_data': {'value': int64_array([1, 3, 30, 30]), 'kind': 'data'},
+                    }
+
+
+class ShapeOfConstFoldingTests(unittest.TestCase):
+    def test1(self):
+        graph = build_graph(nodes_attributes,
+                            [('input', 'input_data'),
+                             ('input_data', 'shapeof_input'),
+                             ('shapeof_input', 'shapeof_input_data'),
+                             ('shapeof_input_data', 'mul'),
+                             ('const', 'const_data'),
+                             ('const_data', 'shapeof_const'),
+                             ('shapeof_const', 'shapeof_const_data'),
+                             ('shapeof_const_data', 'mul'),
+                             ('mul', 'mul_data'),
+                             ('mul_data', 'last')],
+                            {
+                                'input': {'shape': int64_array([1, 3, 30, 30])},
+                                'input_data': {'shape': int64_array([1, 3, 30, 30])},
+                                'shapeof_input': {'value': int64_array([1, 3, 30, 30])},
+                                'shapeof_input_data': {'value': int64_array([1, 3, 30, 30])},
+                                'const': {'value': const_value},
+                                'const_data': {'value': const_value},
+                                'shapeof_const': {'value': int64_array([1, 3, 30, 30])},
+                                'shapeof_const_data': {'value': int64_array([1, 3, 30, 30])},
+                                'mul_data': {'value': int64_array([1, 9, 900, 900])},
+                            },
+                            nodes_with_edges_only=True)
+
+        graph_ref = build_graph(nodes_attributes,
+                                [('input', 'input_data'),
+                                 ('input_data', 'shapeof_input'),
+                                 ('shapeof_input', 'shapeof_input_data'),
+                                 ('shapeof_input_data', 'mul'),
+                                 ('new_const_shapeof', 'new_const_shapeof_data'),
+                                 ('new_const_shapeof_data', 'mul'),
+                                 ('mul', 'mul_data'),
+                                 ('mul_data', 'last')],
+                                {
+                                    'input': {'shape': int64_array([1, 3, 30, 30])},
+                                    'input_data': {'shape': int64_array([1, 3, 30, 30])},
+                                    'shapeof_input': {'value': int64_array([1, 3, 30, 30])},
+                                    'shapeof_input_data': {'value': int64_array([1, 3, 30, 30])},
+                                    'new_const_shapeof': {'value': int64_array([1, 3, 30, 30])},
+                                    'new_const_shapeof_data': {'value': int64_array([1, 3, 30, 30])},
```

Comment:
Why the graph contains new data node `new_const_shapeof_data`? The `shapeof_const_data` data node should be kept! Looks like you loose the debug information (tensor name) after running this transformation.